perf/benchmark/security/generate_policies/generate_policies.go

```diff
@@ -36,27 +41,47 @@ type ruleGenerator struct {
 	gen generator
 }
 
+type Jwks struct {
+	Keys []*Jwk `json:"keys"`
+}
+
+type Jwk struct {
+	Kty string `json:"kty"`
+	E   string `json:"e"`
+	N   string `json:"n"`
+}
+
 type SecurityPolicy struct {
-	AuthZ     AuthorizationPolicy `json:"authZ"`
-	Namespace string              `json:"namespace"`
-	PeerAuthN PeerAuthentication  `json:"peerAuthN"`
+	AuthZ        AuthorizationPolicy   `json:"authZ"`
+	Namespace    string                `json:"namespace"`
+	PeerAuthN    PeerAuthentication    `json:"peerAuthN"`
+	RequestAuthN RequestAuthentication `json:"requestAuthN"`
 }
 
 type AuthorizationPolicy struct {
-	Action        string `json:"action"`
-	NumNamespaces int    `json:"numNamespaces"`
-	NumPaths      int    `json:"numPaths"`
-	NumPolicies   int    `json:"numPolicies"`
-	NumPrincipals int    `json:"numPrincipals"`
-	NumSourceIP   int    `json:"numSourceIP"`
-	NumValues     int    `json:"numValues"`
+	Action                string `json:"action"`
+	MatchRequestPrincipal bool   `json:"matchRequestPrincipal"`
+	NumNamespaces         int    `json:"numNamespaces"`
+	NumPaths              int    `json:"numPaths"`
+	NumPolicies           int    `json:"numPolicies"`
+	NumPrincipals         int    `json:"numPrincipals"`
+	NumSourceIP           int    `json:"numSourceIP"`
+	NumValues             int    `json:"numValues"`
+	NumRequestPrincipals  int    `json:"numRequestPrincipals"`
 }
 
 type PeerAuthentication struct {
 	MtlsMode    string `json:"mtlsMode"`
 	NumPolicies int    `json:"numPolicies"`
 }
 
+type RequestAuthentication struct {
+	InvalidToken bool   `json:"invalidToken"`
```

Comment:
Comment what is this `InvalidToken` doing, it's not very obvious from the name.

Reply:
wrote a comment